Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -29,6 +32,153 @@ Status::Status(FirestoreErrorCode code, absl::string_view msg) {
   state_->msg = static_cast<std::string>(msg);
 }
 
+Status Status::FromErrno(int errno_code, absl::string_view msg) {
+  FirestoreErrorCode canonical_code = CodeForErrno(errno_code);
+  return Status{canonical_code,
+                util::StringFormat("%s (errno %s: %s)", msg, errno_code,
+                                   StrError(errno_code))};
+}
+
+FirestoreErrorCode Status::CodeForErrno(int errno_code) {
```

Comment:
I'm a little worried that this overdoes it. There are really only a few `errno` error codes we care about, but this is essentially what the google3-internal code does (except that not all of the codes there are defined on the Apple platforms).

I'm torn about keeping all this. On the one hand it makes it more likely that we'll react more sanely in circumstances we haven't anticipated, but realistically we're only interacting with errno-returning routines during startup and there we treat every error as effectively fatal.

This seems like a bunch of portability headache for something that doesn't really matter.

Thoughts?